docs/examples/check-email-for-intent.md

```diff
@@ -0,0 +1,243 @@
+---
+layout: default
+title: Check email for Intent
+nav_order: 12
+parent: Examples
+---
+
+# Introduction
+We're going to be creating two workflows one will send an email with a cloud event to trigger the second workflow. The second workflow will read the contents of the email and use AI to work out its intent and respond back if it is negative.
+
+
+## Send an Email and Trigger an Event
+
+```yaml
+id: send-email-trigger-event
+functions:
+- id: smtp
+  image: vorteil/smtp:v3
+- id: request
+  image: vorteil/request:v5
+description: This workflow sends an email and triggers an event.
+states:
+  # continued in next code block
+```
+
+### Send Email
+This action state uses the smtp container to send an email.
+
+```yaml
+- id: sendemail
+  type: action
+  action:
+    function: smtp
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    input:
+      to: "{{.secrets.EMAIL_NAME}}"
+      subject: "Your Review"
+      message: "You are very bad at doing work."
+      from: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      server: smtp.gmail.com
+      port: 587
+  transition: sendcloudevent
+```
+
+### Trigger Event
+This action state uses the request container to send a cloud event back to Direktiv.
+
+```yaml
+- id: sendcloudevent
+  type: action
+  action: 
+    function: request
+    input:
+      method: "POST"
+      url: "http://192.168.1.30/api/namespaces/direktiv/event"
+      body: 
+        id: "read-email-message"
+        specversion: "1.0"
+        type: "smtp"
+```
+
+### Workflow Send Email 
+
+```yaml
+id: send-email-trigger-event
+functions:
+- id: request
+  image: vorteil/request:v5
+- id: smtp
+  image: vorteil/smtp:v3
+description: This workflow sends an email and triggers an event.
+states:
+- id: sendemail
+  type: action
+  action:
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    function: smtp
+    input:
+      to: "{{.secrets.EMAIL_NAME}}"
+      subject: "Your Review"
+      message: "You are very bad at doing work."
+      from: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      server: smtp.gmail.com
+      port: 587
+  transition: sendcloudevent
+- id: sendcloudevent
+  type: action
+  action: 
+    function: request
+    input:
+      method: "POST"
+      url: "http://192.168.1.30/api/namespaces/direktiv/event"
+      body: 
+        id: "read-email-message"
+        specversion: "1.0"
+        type: "smtp"
+```
+
+
+## Read the Email and Check its Intent
+
+```yaml
+id: listen-for-email
+description: This workflow reads an email when a cloud event is received.
+functions:
+- id: imap
+  image: vorteil/imap:v1
+- id: smtp
+  image: vorteil/smtp:v3
+- id: sentiment
+  image: vorteil/google-sentiment-check:v2
+start:
+  type: event
+  state: read-email
+  event:
+    type: smtp
+states:
+  # continued in next code block
+```
+
+**NOTE: when the 'direktiv' namespace gets a cloud event of 'smtp' type this workflow will be triggered.**
```

Comment:
**NOTE: ** This workflow will be triggered when the 'direktiv' namespace receives a cloud event of type `smtp`.